keps/sig-node/277-ephemeral-containers/README.md

```diff
@@ -675,60 +663,113 @@ _This section must be completed when targeting alpha to a release._
+  Component restarts won't affect this feature.
 
 * **What specific metrics should inform a rollback?**
 
+  A rollback is only indicated if there's a catastrophic failure that prevents
+  the cluster from functioning normally, for example if pod or container
+  creation begins to fail.
+
 * **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
   Describe manual testing that was done and the outcomes.
   Longer term, we may want to require automated upgrade/rollback tests, but we
   are missing a bunch of machinery and tooling and can't do that now.
 
+  Since this feature is not critical to production workloads, the main risk
+  is that enabling the feature will adversely affect other components when
+  it becomes enabled by default.
+
+  This feature is implemented in the apiserver and kubelet. The change to the
+  core API has had several releases to soak. For the apiserver, the main risk
+  is described by [Adding Unstable Features to Stable Versions]. Specifically:
+
+  > Ensuring existing data is preserved is needed so that when the feature is
+  > enabled by default in a future version n and data is unconditionally allowed
+  > to be persisted in the field, an n-1 API server (with the feature still
+  > disabled by default) will not drop the data on update.
+
+  We've followed the instructions in this doc for how to persist unstable fields
+  during an update. The case for this feature is slightly more complicated
+  because the field may not be set by the default update resource. This logic
+  has been in place for several releases.
+
+  The apiserver and kubelet have existing upgrade tests
+  (https://testgrid.k8s.io/google-gce-upgrade), but these likely don't exercise
+  ephemeral containers. We'll investigate whether it's possible to add ephemeral
+  containers to these existing tests.
+
+  TODO(verb): create issue to update existing upgrade tests after discussing
+  with PRR reviewer.
+
 * **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
 fields of API types, flags, etc.?**
-  Even if applying deprecation policies, they may still surprise some users.
+
+  No.
 
 ### Monitoring Requirements
 
 _This section must be completed when targeting beta graduation to a release._
 
 * **How can an operator determine if the feature is in use by workloads?**
-  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
-  checking if there are objects with field X set) may be a last resort. Avoid
-  logs or events for this purpose.
+
+  We will create a new gauge metric that's updated during kubelet's reconcile
+  of `v1.Pod` to track the number containers scheduled to this node in the API.
+  This will be slightly different than the existing
+  `kubelet_running_containers`, which describes the kubelet's representation of
+  containers, and will be able to label the metrics with fields that are only
+  available in the API object, such as type of container.
```

Comment:
Are we planning to replace kubelet_running_containers with v1.Pod? It can be handled separately of course.